ethclient/ethclient.go

```diff
@@ -141,6 +143,16 @@ func (ec *client) Client() *rpc.Client {
 
 // Blockchain Access
 
+// ChainConfig retrieves the current chain config.
+func (ec *client) ChainConfig(ctx context.Context) (*params.ChainConfig, error) {
```

Comment:
I meant returning `ChainConfigWithUpgradesJSON` instead of  `ChainConfig` [here](https://github.com/ava-labs/subnet-evm/pull/1040#discussion_r1442719438). Sorry for the confusion.